oasis-reference/src/test/kotlin/com/empowerops/volition/ref_oasis/plugin/PluginServiceFixture.kt

```diff
@@ -0,0 +1,406 @@
+        with(debugLogger.log) {
+            assertThat(size).isEqualTo(1)
+            assertThat(single().sender).isEqualTo(errorResponseDTO.name)
+            assertThat(single().message).isEqualTo("Error update simulation ${errorResponseDTO.name} due to ${errorResponseDTO.message} :\n${errorResponseDTO.exception}")
+        }
+    }
+
+    @Test
+    fun `when request update should send the update but timedout and receive the error`() = runBlocking<Unit> {
+        //setup
+        val inputStream = mock<StreamObserver<RequestQueryDTO>>()
+        val modelService = mock<ModelService> {
+            on { simulations } doReturn listOf(Simulation("Node1", inputStream))
+            on { updateTimeout } doReturn 50//in ms
+        }
+        val eventBus = EventBus()
+        val debugLogger = ConsoleOutput(eventBus)
+        val service = PluginService(modelService, debugLogger, eventBus)
+
+        //act
+        service.onUpdateNodeRequested(SimulationUpdateRequestedEvent("Node1")).join()
+
+        //assert
+        verify(inputStream, times(1)).onNext(check {
+            assertThat(it.hasNodeStatusRequest()).isTrue()
+            assertThat(it.nodeStatusRequest.name).isEqualTo("Node1")
+        })
+
+        verify(modelService, never()).updateSimAndConfiguration(any<Simulation>())
+        with(debugLogger.log) {
+            assertThat(size).isEqualTo(1)
+            assertThat(single().sender).isEqualTo("Optimizer")
+            assertThat(single().message).isEqualTo("Update simulation timeout. Please check simulation is registered and responsive.")
+        }
+    }
+
+    @Test
+    fun `when request update but encounter an exception should log the result`() = runBlocking<Unit> {
+        //setup
+        val inputStream = mock<StreamObserver<RequestQueryDTO>> {
+            on { onNext(any()) } doAnswer { throw testException }
+        }
+        val modelService = mock<ModelService> {
+            on { simulations } doReturn listOf(Simulation("Node1", inputStream))
+            on { updateTimeout } doReturn 500000//in ms
+        }
+        val eventBus = EventBus()
+        val debugLogger = ConsoleOutput(eventBus)
+        val service = PluginService(modelService, debugLogger, eventBus)
+
+        //act
+        service.onUpdateNodeRequested(SimulationUpdateRequestedEvent("Node1")).join()
+
+        //assert
+        verify(modelService, never()).updateSimAndConfiguration(any<Simulation>())
+        with(debugLogger.log) {
+            assertThat(size).isEqualTo(1)
+            assertThat(single().sender).isEqualTo("Optimizer")
+            assertThat(single().message).isEqualTo("Unexpected error happened when update simulation ${responseDTO.name} failed. Please check simulation is registered and responsive. Cause:\n" +
+                    "$testException")
+        }
+    }
+
+    /**
+     * notifyStart/Stop
+     */
+    @Test
+    fun `when notifying start should send the start message to all the registered`() {
```

Comment:
These are covered in system test now but there is no reason not having them here, going to add them